dev/src/dev.clj

```diff
@@ -58,10 +59,28 @@
   (mbc/init!)
   (reset! initialized? true))
 
+(defn- deleted-inmem-databases
+  []
+  (let [h2-dbs (t2/select :model/Database :engine :h2)
+        in-memory? (fn [db] (some-> db :details :db (str/starts-with? "mem:")))
+        can-connect? (fn [db]
+                       (binding [metabase.driver.h2/*allow-testing-h2-connections* true]
+                         (try
+                           (driver/can-connect? :h2 (:details db))
+                           (catch org.h2.jdbc.JdbcSQLNonTransientConnectionException _
+                               false))))]
+    (remove can-connect? (filter in-memory? h2-dbs))))
+
+(defn prune-deleted-inmem-databases!
+  []
+  (doseq [db (deleted-inmem-databases)]
+    (t2/delete! :model/Database :id (:id db))))
```

Comment:
We can delete them all in one delete statement:

```clojure
(when-let [outdated-inmemory (seq (deleted-inmem-databases))]
  (t2/delete! :model/Database :id [:in outdated-inmemory]))
```

Reply:
Very nice!